.github/workflows/validate.yml

```diff
@@ -0,0 +1,23 @@
+name: validate
+on:
+  pull_request:
+    branches:
+      - "**"
+  push:
+    branches:
+      - "main"
+jobs:
+  check-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - run: npm install
+      - run: npm test
+      - name: Check for changes
+        run: |
+          chmod +x "${GITHUB_WORKSPACE}/.github/workflows/validate.sh"
+          "${GITHUB_WORKSPACE}/.github/workflows/validate.sh"
```

Comment:
ni+question: Can the `validate.sh` just be commited as executable (and then it doesn't have to have chmod run on it)? I'm not super familiar with the effect chmod +x when run from different machines. Mostly just curious.

Reply:
Seemed to work. And https://github.com/slackhq/tree-sitter-hack/blob/track-corpus/.github/workflows/validate.sh is marked as an exec file.